pkg/asset/manifests/clusterapi/cluster.go

```diff
@@ -0,0 +1,230 @@
+package clusterapi
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/manifests"
+	"github.com/openshift/installer/pkg/asset/manifests/aws"
+	"github.com/openshift/installer/pkg/asset/manifests/azure"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/asset/openshiftinstall"
+	"github.com/openshift/installer/pkg/asset/rhcos"
+	"github.com/openshift/installer/pkg/clusterapi"
+)
+
+const (
+	manifestDir = "cluster-api-cluster-manifests"
+)
+
+var _ asset.WritableAsset = (*Cluster)(nil)
+
+// Cluster generates manifests for target cluster
+// creation using CAPI.
+type Cluster struct {
+	FileList  []*asset.File
+	Manifests capiutils.Manifests `json:"-"`
```

Comment:
Skipping serialization for `Manifests` is not great ( `json:"-"`) We might have some ways around it. One thing I was considering is whether we could create a new type of asset, which included the runtime object in addition to `asset.File`

Reply:
🤔 Aren't these Manifests only collected to be consumed internally as objects and for ordering? The FileList will still load/store